ui/src/data/reducers/index.ts

```diff
@@ -0,0 +1,35 @@
+// Libraries
+import {produce} from 'immer'
+
+// Actions
+import {Action} from 'src/data/actions'
+
+export interface DataState {
+  queryResultsByQueryID: {[queryID: string]: string[]}
+}
+
+export const initialState: DataState = {
+  queryResultsByQueryID: {},
+}
+
+export const dataReducer = (
+  state: DataState = initialState,
+  action: Action
+): DataState => {
+  switch (action.type) {
+    case 'SET_QUERY_RESULTS_BY_QUERY': {
+      return produce(state, draftState => {
+        const {queryID, files} = action
+        if (queryID && files.length) {
+          draftState.queryResultsByQueryID[queryID] = files
+        }
+      })
+    }
+
+    case 'RESET_CACHED_QUERY_RESULTS': {
+      return initialState
+    }
+  }
+
+  return state
+}
```

Comment:
is this file necessary? why can't the query results be stored in the time machine reducer?

Reply:
I had originally set the data in the time machine, but it became a little tricky accessing the data since timeMachine is partitioned into segmented subsets. The thinking with this feature is that we might be also benefit from this caching across the platform in the future, so if the same query is run in data-explorer and a dashboard cell, we'd be able to cache those results and reduce querying